source/change-streams/tests/README.rst

```diff
@@ -158,4 +158,41 @@ The following tests have not yet been automated, but MUST still be tested
 #. The ``killCursors`` command sent during the "Resume Process" must not be allowed to throw an exception.
 #. ``$changeStream`` stage for ``ChangeStream`` against a server ``>=4.0`` that has not received any results yet MUST include a ``startAtOperationTime`` option when resuming a changestream.
 #. ``ChangeStream`` will resume after a ``killCursors`` command is issued for its child cursor.
-
+#. ``ChangeStream`` against a server ``>=4.2`` must track the latest ``postBatchResumeToken`` included in an ``aggregate`` or ``getMore`` response.
+#. ``ChangeStream`` must resume ``resumeAfter`` value that always agrees with the value which was specified in the ``resumeAfter`` option while executing an ``aggregate`` command.
+#. - For a ``ChangeStream`` under these conditions:
+      - running against a server ``>=4.2``.
+      - the batch is empty or has been iterated to the last document.
+   - Expected result: 
+       - The ``getResumeToken`` must return the ``postBatchResumeToken`` from the current command response.
+#. - For a ``ChangeStream`` under these conditions:
+      - running against a server ``<4.2``.
+      - the batch is empty or has been iterated to the last document.
+   - Expected result: 
+       - The ``getResumeToken`` must return the ``startAfter`` or ``resumeAfter`` from the current command request. 
+       - If ``startAfter`` and ``resumeAfter`` are absent, ``getResumeToken`` result must be empty.
+#. - For a ``ChangeStream`` under these conditions:
+      - the batch is not empty.
+      - the batch has been iterated up to but not including the last element.
+   - Expected result:
+      - The ``getResumeToken`` must return the last cached document resume token.
+#. - For a ``ChangeStream`` under these conditions:
+      - the batch is not empty.
+      - the batch hasn’t been iterated at all.
+      - executing an ``aggregate`` command.
+   - Expected result:
+      - The ``getResumeToken`` must return the ``startAfter`` or ``resumeAfter`` from the initial ``aggregate`` request.
+#. - For a ``ChangeStream`` under these conditions:
+      - running against a server ``>=4.2``.
+      - the batch is not empty.
+      - the batch hasn’t been iterated at all.
+      - executing a ``getMore`` command.
+   - Expected result:
+      - The ``getResumeToken`` must return the ``postBatchResumeToken`` from the previous command response.
+#. - For a ``ChangeStream`` under these conditions:
+      - running against a server ``<4.2``.
+      - the batch is not empty.
+      - the batch hasn’t been iterated at all.
+      - executing a ``getMore`` command.
+   - Expected result:
+      - The ``getResumeToken`` must return the ``startAfter`` or ``resumeAfter`` from the previous command request.
```

Comment:
In this case, we should use the _id of the last document returned (from the previous batch). If there are no previous documents returned, then we return ``startAfter``/``resumeAfter`` of the initial aggregate. And if that wasn't specified, we return nothing.